.build/build.ps1

```diff
@@ -61,46 +53,42 @@ task Clean {
     }
 }
 
+task Compile {
+    Write-Info "Running 'dotnet build'"
 
-# RestorePackages task, restores all the NuGet packages.
-task RestorePackages {
-    Write-Info "Restoring NuGet packages for solution $SolutionPath"
-
-    & $NuGetPath @('restore', $SolutionPath)
-}
-
-
-# Compile task, runs MSBuild to build the solution.
-task Compile  RestorePackages, {
-    Write-Info "Compiling solution $SolutionPath"
-
-    exec {
-        msbuild "$SolutionPath" /nodeReuse:False /target:Build /property:Configuration=Release
+    Push-Location -Path $SourceDir
+    try {
+        exec {
+            dotnet build 
+        }
+    } finally {
+        Pop-Location
     }
 }
 
 
-# Test task, runs the automated tests.
+# Test task, builds the solution and runs the automated tests.
 task Test  Compile, {
-    Write-Info 'Running tests'
-
-    $TestAssemblyPath = "$SourceDir\Ulibs.Tests\bin\$Configuration\ULibs.Tests.dll" | Resolve-Path
-    Invoke-NUnit3ForAssembly -AssemblyPath $TestAssemblyPath `
-                             -NUnitVersion '3.8.0' `
-                             -FrameworkVersion 'net-4.0' `
-                             -EnableCodeCoverage $True `
-                             -DotCoverFilters '+:ULibs.*;-:*.Tests' `
-                             -DotCoverAttributeFilters '*.ExcludeFromCodeCoverageAttribute'
-                                 
-    $CoverageResultsPath = "$TestAssemblyPath.TestResult.coverage.snap" | Resolve-Path
-    TeamCity-ImportDotNetCoverageResult 'dotcover' $CoverageResultsPath        
+    Write-Info "Running 'dotnet test'"
+
+    Push-Location -Path $SourceDir
+    try {
+        exec {
+            dotnet test

```

Comment:
It looks like this will run the tests on whatever the build agent framework is - do we want to try explicitly run the tests on both .net core and framework?

Reply:
It runs the tests for all of the targeted frameworks. Right now, that's net461 and netcoreapp3.1. :)

I've already updated the TC build configurations to require Windows and dotnet cli v3.1 or later.

Reply:
We could try to run the tests on Linux, but that would require a lot more TC jiggery pokery, and messing with artifacts. I don't think we have anything in these libraries that makes it worth the effort right now. Maybe later, though.

Reply:
Well, to be clear, even though we target net461 and netcoreapp3.1, the tests will run under whatever later compatible versions are actually installed on the build agents. This is largely true for all of our tests across TC. I'm not sure we really care about that discrepancy?

Reply:
> It runs the tests for all of the targeted frameworks. Right now, that's net461 and netcoreapp3.1. :)

Oh cool, I didn't realise this :) I got a bit confused since teamcity is only reporting one run for each test - normally it shows up as multiple results per test name if tests get rerun with the same name

Reply:
![image](https://user-images.githubusercontent.com/382638/77746608-1197b300-7015-11ea-90a9-509add89c211.png)

That might be down to the TC logger not disambiguating between the same tests running under different runtimes. But the command itself does indeed run the tests for both targets.

Reply:
Oh, looking at TC, it reports a test count of 867, but this is different to the 3064 test runs you see from my screenshot above. I think the difference is down to tests vs test runs. If you look closer in TC, each of those tests actually runs multiple times. About 4 times on average, at a quick glance.
https://buildserver.red-gate.com/viewLog.html?buildId=12163457&buildTypeId=SharedComponents_MicroLibraries&tab=testsInfo

Reply:
Aha! I think this is a problem with the new teamcity UI. Here's what I was seeing:
<img width="1034" alt="image" src="https://user-images.githubusercontent.com/112905/77747279-4a845780-7016-11ea-8a03-cab295ca4861.png">

and here's what I was expecting to see:
<img width="1282" alt="image" src="https://user-images.githubusercontent.com/112905/77747336-6982e980-7016-11ea-8b0c-72b4847075e7.png">

Reply:
Ah, yeah. I'm still using the old UI. I'm going to give it 6 months or so for them to iron out all of these glitches before I switch.